server/routes/bookingsRoute.js

```diff
@@ -47,4 +48,49 @@ router.post('/students/:student_id/bookings', lectureValidation.checkLecture(),v
         res.status(400).json(error);
     }
 })
+
+
+/**
+ * @swagger
+ * //students/{student_id}/delete_booking:
+ *  post:
+ *    tags:
+ *      - students
+ *    summary: "Delete a new booking"
+ *    description: "A student can delete a lecture "
+ *    parameters:
+ *      - in: path
+ *        name: student_id
+ *        required: true
+ *        schema:
+ *          type: integer
+ *          format: int64
+ *    responses:
+ *       "201 and num=1":
+ *         description: "Successful deletion"
+ *         schema:
+ *           type: "object"
+ *       "201 and num=0":
+ *         description: "Nothing to delete"
+ *         schema:
+ *           type: "object"
+ *       "400":
+ *         description: "Invalid status value"
+ *    security:
+ *     - petstore_auth:
+ *       - "write:pets"
+ *       - "read:pets"
+ */
+router.post('/students/:student_id/delete_booking', lectureValidation.checkLecture(),validator, async(req,res) =>{
+    const {lecture_id} = req.body;
+    const student_id= +req.params.student_id;
+    try{
+        let number = await bookingService.deleteBooking({lecture_id,student_id});
+        let booking={number:number};
+        return res.status(201).json(booking);
```

Comment:
I think that it's correct that the dao and the service return the number of affected rows.. anyway, maybe the route could just return two different HTTP codes based on that number, with an empty response body.

PS: `201 Created` is not the best return value I suppose, we can just opt for `200 OK` here